File: ansible/generate_inventory.py
import subprocess
import re

output = subprocess.check_output([
    "docker", "ps", "--filter", "name=target", "--format", "{{.Names}} {{.Networks}}"
])
lines = output.decode().splitlines()

host_map = {}
for line in lines:
    parts = line.split()
    name = parts[0]
    inspect = subprocess.check_output(["docker", "inspect", name])
    ip_match = re.search(r'"IPAddress": "([0-9\.]+)"', inspect.decode())
    if ip_match:
        ip = ip_match.group(1)
        host_map[name] = ip

with open("ansible/hosts", "w") as f:
    f.write("[docker-hosts]\n")
    for idx, (name, ip) in enumerate(host_map.items(), 1):
        f.write(f"target_node{idx} ansible_host={ip}\n")
    f.write("\n[docker-hosts:vars]\nansible_user=ansible\n")

File: generate_inventory.js
const { execSync } = require("child_process");
const fs = require("fs");

// Get container names and networks for containers matching "target"
const psOutput = execSync(
  `docker ps --filter name=target --format "{{.Names}}"`
)
  .toString()
  .trim();

if (!psOutput) {
  console.log("No target containers found.");
  process.exit(0);
}

const containerNames = psOutput.split("\n");

const hostMap = {};

containerNames.forEach((name) => {
  const inspectOutput = execSync(`docker inspect ${name}`).toString();
  const ipMatch = inspectOutput.match(/"IPAddress":\s*"([0-9.]+)"/);
  if (ipMatch) {
    hostMap[name] = ipMatch[1];
  }
});

// Generate hosts file content
let content = "[docker-hosts]\n";
let idx = 1;
for (const [name, ip] of Object.entries(hostMap)) {
  content += `${name} ansible_host=${ip}\n`;
  idx++;
}

content += "\n[docker-hosts:vars]\n";
content += "ansible_user=ansible\n";

// Write to ansible/hosts
fs.writeFileSync("ansible/hosts", content);
